File: zmq/farmer.rb
#!/usr/bin/env ruby

require 'ffi-rzmq'
require './const'

context = ZMQ::Context.new
me  = ARGV[0]
s   = context.socket(ZMQ::PUSH)         	# create a push socket
src = (me == '1')? SRC1  : SRC2         	# check task source host
prt = (me == '1')? PORT1 : PORT2        	# check task source port
p   = "tcp://#{src}:#{prt}"             	# how and where to connect
s.bind(p)                               	# bind socket to address

100.times do                            	# generate 100 workloads
  workload = rand(1..100)                     	# compute workload
  s.send_string(Marshal.dump([me,workload]))	# send workload to worker
end

File: zmq/farmer.py
#!/usr/bin/env python3

import zmq, time, pickle, sys, random
from const import *

context = zmq.Context()
me  = str(sys.argv[1])
s   = context.socket(zmq.PUSH)          # create a push socket
src = SRC1  if me == '1' else SRC2      # check task source host
prt = PORT1 if me == '1' else PORT2     # check task source port
p   = f"tcp://{src}:{prt}"              # how and where to connect
s.bind(p)                               # bind socket to address

for i in range(100):                    # generate 100 workloads
  workload = random.randint(1, 100)     # compute workload
  s.send(pickle.dumps((me,workload)))   # send workload to worker

